### PR TITLE
Send team welcome chat message in bg

### PR DESCRIPTION
**Pull request by mlsteele**
Put the chat send in the background. It took 800ms out of 2.2s for `TeamAddMember` in one instance.

### CODE REVIEW

**Comment by mmaxim** [comment]
Why did it take 800ms? That seems very high.

File: go/service/teams.go [hunk 1]
```diff
@@ -222,8 +222,17 @@ func (h *TeamsHandler) TeamAddMember(ctx context.Context, arg keybase1.TeamAddMe
...
 
-	result.ChatSent = teams.SendTeamChatWelcomeMessage(ctx, h.G().ExternalG(), arg.Name,
-		result.User.Username)
+	result.ChatSending = true
+	go func() {
+		ctx := libkb.WithLogTag(context.Background(), "BG")
+		err := teams.SendTeamChatWelcomeMessage(ctx, h.G().ExternalG(), arg.Name,
+			result.User.Username)
+		if err != nil {
+			h.G().Log.CWarningf(ctx, "send team welcome message: %v", err)
```
Comment (mmaxim) [inline go/service/teams.go]: I would make this `Debug`. If it is in a goroutine its not like the user will see it as part of their CLI experience anyway, it is just going to show up at some point in time.

File: go/teams/chatmessaging.go [hunk 2]
```diff
@@ -2,25 +2,18 @@ package teams
...
 
-func SendTeamChatWelcomeMessage(ctx context.Context, g *libkb.GlobalContext, team, user string) (res bool) {
-	var err error
-	defer func() {
-		if err != nil {
-			g.Log.CWarningf(ctx, "failed to send team welcome message: %s", err.Error())
-		}
-	}()
-
+func SendTeamChatWelcomeMessage(ctx context.Context, g *libkb.GlobalContext, team, user string) (err error) {
```
Comment (mmaxim) [inline go/teams/chatmessaging.go]: Ah I see you did all this for the `Warning` above. I don't think it makes any sense to do it that way.

**Comment by mlsteele** [comment]
[We'll see](https://github.com/keybase/client/pull/12799)

**Comment by mlsteele** [comment]
Seems fine. The other PR passed and it doesn't send the chat at all, so nothing is testing this.

**Pull request merged**